### PR TITLE
Rewriting api

### PR DESCRIPTION
**Pull request by SodaVolcano**
The current API is not very helpful and I found it a bit confusing so I made small changes to the routes.

- Added getting tasks/bids specific to a user
- Get bid specific to a task
- Added `/bids/` and `users/` routes

This branch was originally #156 but there's no API available so I worked on the backend for a bit.

**WARNING** - API calls on the front end have not been changed (only relevant if any pages made calls to the old API)

### CODE REVIEW

**Review by ErikaKK — COMMENTED** [review]
I actually have a question. For the task table,  I don't know the task_id(which should be auto increment) when I create a new task, so does the name of the route make sense if I have to know the task_id when I send post req.

**Comment by ErikaKK** [comment]
And I already had an error when I add call api in creating new task
image

That means the primary key is wrong?

**Comment by SodaVolcano** [comment]
> I actually have a question. For the task table, I don't know the task_id(which should be auto increment) when I create a new task, so does the name of the route make sense if I have to know the task_id when I send post req.

The new routes all only handle `GET` requests, you should be able to make new tasks by making a POST request to `tasks/` because of this line
```python
router.register(r'tasks', TasksViewSet, basename='tasks')
```

**Comment by SodaVolcano** [comment]
> And I already had an error when I add call api in creating new task image
> 
> That means the primary key is wrong?

How are you calling the api? I just go to the links directly and they seem to work. See http://localhost:8000/api/app/users/8/bids
and 
http://localhost:8000/api/app/users/8/tasks

File: client/src/pages/signin-test.tsx [hunk 1]
```diff
@@ -37,9 +37,9 @@ const TestPostRequestPage = () => {
   const handleTestRegisterRequest = () => {
     axios
       .post("http://localhost:8000/api/app/register/", {
-        email: "user123@example.com",
+        email: "user1234567il3@example.com",
         password: "user123abc!@#",
-        username: "user123",
+        username: "useerghjr123",
```
Comment (yunho7687) [inline client/src/pages/signin-test.tsx:42]: any reason for changing this?

**Review by ErikaKK — APPROVED** [review]

**Review by yunho7687 — COMMENTED** [review]
amazing work! will approve after you reply my question!

File: client/src/pages/signin-test.tsx (hunk 1 above)
Comment (SodaVolcano) [reply client/src/pages/signin-test.tsx:42]: > any reason for changing this? 

I actually don't remember why or how I changed it. I probably was trying to test what it's used for or something? That whole file should be deleted actually. On the original branch I implemented functional API calls so there's no need for this, I'll remove it

Comment (SodaVolcano) [reply client/src/pages/signin-test.tsx:42]: Wait no, idk what this page is used for. Probably leave it on main for now and we can delete all test pages later once the APIs calls are implemented 

How did I edit this file???

**Pull request merged**